python/cudf/cudf/pylibcudf_tests/common/utils.py

```diff
@@ -102,14 +104,56 @@ def _make_fields_nullable(typ):
             return pa.list_(new_fields[0])
         return typ
 
+    def _contains_type(parent_typ, typ_checker):
+        """
+        Check whether the parent or one of the children
+        satisfies the typ_checker.
+        """
+        if typ_checker(parent_typ):
+            return True
+        if pa.types.is_nested(parent_typ):
+            for i in range(parent_typ.num_fields):
+                if _contains_type(parent_typ.field(i).type, typ_checker):
+                    return True
+        return False
+
     if not check_field_nullability:
         rhs_type = _make_fields_nullable(rhs.type)
         rhs = rhs.cast(rhs_type)
 
         lhs_type = _make_fields_nullable(lhs.type)
-        lhs = rhs.cast(lhs_type)
+        lhs = lhs.cast(lhs_type)
 
-    assert lhs.equals(rhs)
+    assert lhs.type == rhs.type, f"{lhs.type} != {rhs.type}"
+    if _contains_type(lhs.type, pa.types.is_floating) and _contains_type(
+        rhs.type, pa.types.is_floating
+    ):
+        # Flatten to do comparisons if nested
+        def _flatten_arrays(arr):
+            flat_arrs = []
+            if pa.types.is_nested(arr.type):
+                flattened = arr.flatten()
+                if isinstance(flattened, list):
+                    for arr in flattened:
+                        flat_arrs += _flatten_arrays(arr)
+                else:
+                    flat_arrs = [flattened]
+            else:
+                flat_arrs = [arr]
+            return flat_arrs
```

Comment:
```suggestion
            if pa.types.is_nested(arr.type):
                flattened = arr.flatten()
                if isinstance(flattened, list):
                    return [_flatten_arrays(arr) for arr in flattened]
                else:
                    return [flattened]
            else:
                return [arr]
```

Reply:
This makes nested lists which isn't what I want.

(I'd like to have all the nested array values together in one list, and compare each array one by one).